test/org/zaproxy/zap/extension/pscanrules/CookieHttpOnlyScannerUnitTest.java

```diff
@@ -0,0 +1,127 @@
+        assertThat(alertsRaised.get(0).getParam(), equalTo("test"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("Set-Cookie: test"));
+    }
+    
+    @Test
+    public void noCookie() throws HttpMalformedHeaderException {
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+    
+    @Test
+    public void httpOnlyAttribute() throws HttpMalformedHeaderException {
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Set-Cookie: test=123; Path=/; HttpOnly\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+    
+    @Test
+    public void secondCookieNoHttpOnlyAttribute() throws HttpMalformedHeaderException {
+        
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+        
+        msg.setResponseBody("<html></html>");
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n" +
+                "Server: Apache-Coyote/1.1\r\n" +
+                "Set-Cookie: hasatt=test123; Path=/; HttpOnly\r\n" +
+                "Set-Cookie: test=123; Path=/;\r\n" +
+                "Content-Type: text/html;charset=ISO-8859-1\r\n" +
+                "Content-Length: " + msg.getResponseBody().length() + "\r\n");
+        String response = msg.getResponseHeader().toString() + msg.getResponseBody().toString();
+        Source source = new Source(response);
+        rule.scanHttpResponseReceive(msg, -1, source);
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("test"));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("Set-Cookie: test"));
+    }
+    
```

Comment:
OK, supporting this looks like it could be a nightmare to parse correctly.
And the latest RFC https://tools.ietf.org/html/rfc6265#section-3 states that "Origin servers SHOULD NOT fold multiple Set-Cookie header fields into a single header field." On this basis I'm proposing that we dont support it, at least not at this stage. This is not a regression, we didnt support it before;)